### ISSUE TITLE
(dump-unsat-cores, incremental) Non-deterministic segfault on QF_LRA formula

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5079]
Hi,
For this formula, CVC4  throws out a segmentation fault:
```
[634] % cvc4 small.smt2
sat
sat
unsat
[635] % cvc4 small.smt2
sat
sat
CVC4 suffered a segfault.
Offending address is 0x7faa1777300c
Aborted
[636] % cvc4 small.smt2
sat
sat
CVC4 suffered a segfault.
Offending address is 0x7f9d0688f00c
Aborted
[637] % 
[637] % cat small.smt2
(set-logic ALL)
(set-option :dump-unsat-cores true)
(set-option :incremental true)
(declare-fun a () Real)
(declare-fun b () Real)
(assert (<= b 0.25))
(assert (= (- b a) 3))
(assert (or (> (* 2 b) 0) (= (/ 1 b) 3)))
(check-sat)
(push)
(check-sat)
(pop)
(assert (>= (* 3 a) 1))
(check-sat)
[638] %
```

OS: Ubuntu 18.04
Commit: 70bfc4f

**Comment by nafur** [comment]
In debug configuration, I see 
```
Fatal failure within CVC4::ClauseId CVC4::TSatProof<Solver>::resolveUnit(typename Solver::TLit) [with Solver = CVC4::Minisat::Solver; CVC4::ClauseId = unsigned int; typename Solver::TLit = CVC4::Minisat::Lit] at /home/gereon/cvc4_proof/src/./proof/sat_proof_implementation.h:795
Check failure

 reason_ref != Solver::TCRef_Undef

Aborted (core dumped)
```
on
```
(set-option :dump-unsat-cores true)
(set-option :incremental true)
(declare-fun a () Real)
(declare-fun b () Real)
(assert (< b 0.2))
(assert (= 1 (- b a)))
(assert (or (> b 0) (= 1 (/ 1 b))))
(check-sat)
(push)
(check-sat)
(pop)
(assert (= a 1))
(check-sat)
```

**Comment by HanielB** [comment]
As an internal note, this is works on proof-new. When we start using it for unsat-cores we will close this issue.

**Comment by HanielB** [comment]
Closed via #6574
